src/main/java/org/gridsuite/voltageinit/server/service/VoltageInitService.java

```diff
@@ -82,28 +110,62 @@ public UUID runAndSaveResult(UUID networkUuid, String variantId, String receiver
+        } else if (Double.isNaN(voltageLevel.getHighVoltageLimit()) && isHighVoltageLimitDefaultSet) {
+            specificVoltageLimits.add(new VoltageLimitOverride(voltageLevel.getId(), VoltageLimitType.HIGH_VOLTAGE_LIMIT, false, voltageLevelDefaultLimits.get(voltageLevel.getId()).getHighVoltageLimit() + (isHighVoltageLimitModificationSet ? voltageLevelModificationLimits.get(voltageLevel.getId()).getHighVoltageLimit() : 0)));
+        }
+    }
+
+    public OpenReacParameters buildOpenReacParameters(Optional<VoltageInitParametersEntity> voltageInitParametersEntity, UUID networkUuid, String variantId) {
         OpenReacParameters parameters = new OpenReacParameters();
         List<VoltageLimitOverride> specificVoltageLimits = new ArrayList<>();
         List<String> constantQGenerators = new ArrayList<>();
         List<String> variableTwoWindingsTransformers = new ArrayList<>();
         List<String> variableShuntCompensators = new ArrayList<>();
+
         voltageInitParametersEntity.ifPresent(voltageInitParameters -> {
             if (voltageInitParameters.getVoltageLimits() != null) {
-                voltageInitParameters.getVoltageLimits().forEach(voltageLimit -> {
-                    var filterEquipments = filterService.exportFilters(voltageLimit.getFilters().stream().map(FilterEquipmentsEmbeddable::getFilterId).toList(), networkUuid, variantId);
-                    filterEquipments.forEach(filterEquipment ->
-                            filterEquipment.getIdentifiableAttributes().forEach(idenfiableAttribute -> {
-                                    specificVoltageLimits.add(new VoltageLimitOverride(idenfiableAttribute.getId(), VoltageLimitType.LOW_VOLTAGE_LIMIT, true, voltageLimit.getLowVoltageLimit()));
-                                    specificVoltageLimits.add(new VoltageLimitOverride(idenfiableAttribute.getId(), VoltageLimitType.HIGH_VOLTAGE_LIMIT, true, voltageLimit.getHighVoltageLimit()));
-                                }
-                            )
-                    );
-                });
-                constantQGenerators.addAll(toEquipmentIdsList(voltageInitParameters.getConstantQGenerators(), networkUuid, variantId));
-                variableTwoWindingsTransformers.addAll(toEquipmentIdsList(voltageInitParameters.getVariableTwoWindingsTransformers(), networkUuid, variantId));
-                variableShuntCompensators.addAll(toEquipmentIdsList(voltageInitParameters.getVariableShuntCompensators(), networkUuid, variantId));
+                Network network = getNetwork(networkUuid, PreloadingStrategy.COLLECTION, variantId);
+
+                Map<String, VoltageLimitEntity> voltageLevelDefaultLimits = resolveVoltageLevelLimits(voltageInitParameters.getVoltageLimits().stream().filter(voltageLimit -> VoltageLimitParameterType.DEFAULT.equals(voltageLimit.getVoltageLimitParameterType())).toList(), networkUuid, network.getVariantManager().getWorkingVariantId());
```

Comment:
why use the same entity for default values and modifications ?

Reply:
I've done so because they share the same data structure and conceptually they are both about voltage limits, adding an enum to distinguish their usage sounded good to me